### ISSUE TITLE
Backward compatibility on the api | non go module | go1.11.1 version 

### ISSUE THREAD

**Issue by shivakumarss** [stretchr/testify#981]
Hi,

I am using the go project which is built on go1.11.1 and the same built without using modules. It uses GOHOME.

this project is part of code integration.  As part of building the project I am using the following command to build.

```
go get github.com/stretchr/testify/assert
```
After which i get an exception 
```
# github.com/stretchr/testify/assert
../../stretchr/testify/assert/assertions.go:1703:5: undefined: errors.Is
../../stretchr/testify/assert/assertions.go:1726:6: undefined: errors.Is
../../stretchr/testify/assert/assertions.go:1749:5: undefined: errors.As
../../stretchr/testify/assert/assertions.go:1766:7: undefined: errors.Unwrap
../../stretchr/testify/assert/assertions.go:1770:7: undefined: errors.Unwrap
```

I found a work around for my local box, i have to go to `${GOHOME}/src/github.com/stretchr/testify` and downgrade to  `v1.4.0` by `git checkout v1.4.0` which will work fine.

Could you please help me handling this case in the CI tool. 
is there any way i can checkout specific version while getting the binaries something like `go get github.com/stretchr/testify/assert@v1.4.0`  I tried it but unfortunately it doesn't seems to work on non-module go package.
 

Thanks.

**Comment by eking-isp** [comment]
I am encountering the same problem, I will note that we also cannot use gopkg.in links to solve this because that only allows incrementing on major version number.

**Comment by boyan-soubachov** [comment]
Hi,

We dropped support for pre-1.13 Go language support recently on _master_ (as per the Go team's release policy). We however still haven't cut the release (it will happen only at the end of the year) to give teams a chance to upgrade & change.

If you pin and pull the latest tagged release (1.6.1) your issue should be resolved.

**Comment by shivakumarss** [comment]
I got this resolved after upgrading go version to `1.14.1`. CI is fine now.
